text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
>You save an additional allocation by not producing a vector.

And like I said, that would matter if you didn't allocate for each argument. The cost for allocating the vector should be irrelevant compared to the cost for allocating the arguments. What containers other than a vector are you planning to collect the arguments into?

>Seems worth looking at, though I believe on Windows at least we must always copy

Which does not imply that a call to args has to return owning structures.